templates/types/streaming/fastapi/app/api/routers/chat.py

```diff
@@ -40,12 +45,35 @@ async def chat(
         )
         event_handler = EventCallbackHandler()
         chat_engine = get_chat_engine(
-            filters=filters, params=params, event_handlers=[event_handler]
+            filters=filters,
+            params=params,
+            event_handlers=[event_handler],
+            chat_history=messages,
         )
-        response = await chat_engine.astream_chat(last_message_content, messages)
-        process_response_nodes(response.source_nodes, background_tasks)
 
-        return VercelStreamResponse(request, event_handler, response, data)
+        if isinstance(chat_engine, CondensePlusContextChatEngine) or isinstance(
```

Comment:
this condition is confusing for people getting started (chat.py is the file we're telling users to use).
in nextjs we're copying, therefore a special `route.ts` for workflow and a special one for normal chat engine. 
this seems easier to understand for me